topology/m4/eq_fir.m4

```diff
@@ -0,0 +1,71 @@
+divert(-1)
+
+dnl Define macro for Eq effect widget
+
+dnl EQ name)
+define(`N_EQ_FIR', `EQFIR'PIPELINE_ID`.'$1)
+
+dnl W_EQ(name, format, periods_sink, periods_source, preload, kcontrols_list)
+define(`W_EQ_FIR',
+`SectionVendorTuples."'N_EQ_FIR($1)`_tuples_w" {'
+`	tokens "sof_comp_tokens"'
+`	tuples."word" {'
+`		SOF_TKN_COMP_PERIOD_SINK_COUNT'		STR($3)
+`		SOF_TKN_COMP_PERIOD_SOURCE_COUNT'	STR($4)
+`		SOF_TKN_COMP_PRELOAD_COUNT'		STR($5)
+`	}'
+`}'
+`SectionData."'N_EQ_FIR($1)`_data_w" {'
+`	tuples "'N_EQ_FIR($1)`_tuples_w"'
+`}'
+`SectionVendorTuples."'N_EQ_FIR($1)`_tuples_str" {'
+`	tokens "sof_comp_tokens"'
+`	tuples."string" {'
+`		SOF_TKN_COMP_FORMAT'	STR($2)
+`	}'
+`}'
+`SectionData."'N_EQ_FIR($1)`_data_str" {'
+`	tuples "'N_EQ_FIR($1)`_tuples_str"'
+`}'
+`SectionVendorTuples."'N_EQ_FIR($1)`_tuples_str_type" {'
+`	tokens "sof_effect_tokens"'
+`	tuples."string" {'
+`		SOF_TKN_EFFECT_TYPE'	"EQFIR"
+`	}'
+`}'
+`SectionData."'N_EQ_FIR($1)`_data_str_type" {'
+`	tuples "'N_EQ_FIR($1)`_tuples_str_type"'
+`}'
+`SectionData."'N_EQ_FIR($1)`_data_bytes" {'
+`	bytes "0x00,0x00,0x00,0x04,0x00,0x02,0x00,0x01,0x00,0x01,0x00,'
```

Comment:
IIRC SectionData can also use files as input (i.e. it can read binary coefficients), this way we could specify the filename to the N_EQ_FIR as an argument.

Reply:
Yes we discussed with Seppo this would be better/nicer than copying octave script output here. I tested it but for some reason could not get the env var working in my shell, so alsatplg was looking the file from some /usr/foo/bar/etc. But yeah,  that was the idea.

Reply:
ok, we need to patch alsatplg, sounds like a bug - we should be able to use an ENV variable or cmd line  to specify "include" folders for files. This will need upstreamed to alsa dev mailing list and will need commit ID updated on our Dockerfile once merged.

Reply:
ok, this was just my stupidity, now got the file working with with ALSA_CONFIG_TPLG=.
I still have to test this, but will probably update this method